### DISCUSSION TITLE
device identification in the network

### DISCUSSION THREAD

**Discussion by RadutoKomarotti** [lathoub/Arduino-AppleMIDI-Library#165] [General]
Hello,

i have a question i am a beginner and i would like to run rtp midi on arduino. I tried your NoteOnOffEverySec example code. It works. My question is, does the Arduino as a RTP midi device always have to be entered manually? (name, address, port) Or is there a possibility that it would appear by itself on the MIDI Network setup - Directory tab like other devices, e.g. iPhone, and it was enough to connect it with a button? I would like to connect to a device that works with rtp midi, but it does not have the option to enter a manual address and port.

Thank you

**Comment by lathoub** [comment]
Devices can announce themselves using Bonjour and they will show up in the `MIDI Network Setup`.

I use:
https://github.com/lathoub/EthernetBonjour

**Reply by RadutoKomarotti** [reply]
Thank you for the advice. Could I ask for some code (example)??

Thank you

**Comment by krystianlenc** [comment]
Myślę, że rozwiązanie dla Ciebie to funkcja sendinvite (po ethernet działa świetnie i łączy się z moim iphone bez konieczności pośrednictwa macbook.

niestety kompiluje się tylko z sieci Ethernet, podczas korzystania z Wi-Fi pojawia się błąd kompilacji „Błąd kompilacji: „klasa appleMidi::AppleMIDISession” nie ma elementu o nazwie „sendInvite””

Jeśli używasz ethernet, wszystko powinno zadziałać, co do wifi nie wiem co może być problemem, proszę @lathoub o pomoc!

**Comment by RadutoKomarotti** [comment]
... I'm not very good at programming. I am unable to compile the code that would ensure the visibility of the device on the network. Does anyone have any working code available? Just to verify the functionality, then I would modify it for my needs. (I use ethernet shield w5100)
Thanks

**Comment by RadutoKomarotti** [comment]
Hello,

I tried some code that you had the AI ​​build. Next, I tried the trial code from Github Ethernet Bonjour, set up the Discovery bonjour monitor ... still can't see the device! It appeared once for about a second and then disappeared again, that's all. I do not know what do with it .. :(

Couldn't there still be a problem in the router? I have the Arduino connected to the router via LAN and the Mac to the same via WIFI. ???
